// react-native-audio-api/lib/module/core/AudioDecoder.js
"use strict";

import AudioBuffer from "./AudioBuffer.js";
class AudioDecoder {
  static instance = null;
  constructor() {
    this.decoder = global.createAudioDecoder();
  }
  static getInstance() {
    if (!AudioDecoder.instance) {
      AudioDecoder.instance = new AudioDecoder();
    }
    return AudioDecoder.instance;
  }
  async decodeAudioDataInstance(input, sampleRate) {
    let buffer;
    if (typeof input === 'string') {
      // Remove the file:// prefix if it exists
      if (input.startsWith('file://')) {
        input = input.replace('file://', '');
      }
      buffer = await this.decoder.decodeWithFilePath(input, sampleRate ?? 0);
    } else if (input instanceof ArrayBuffer) {
      buffer = await this.decoder.decodeWithMemoryBlock(new Uint8Array(input), sampleRate ?? 0);
    }
    if (!buffer) {
      throw new Error('Unsupported input type or failed to decode audio');
    }
    return new AudioBuffer(buffer);
  }
  async decodePCMInBase64Instance(base64String, inputSampleRate, inputChannelCount, interleaved) {
    const buffer = await this.decoder.decodeWithPCMInBase64(base64String, inputSampleRate, inputChannelCount, interleaved);
    return new AudioBuffer(buffer);
  }
}
export async function decodeAudioData(input, sampleRate) {
  return AudioDecoder.getInstance().decodeAudioDataInstance(input, sampleRate);
}
export async function decodePCMInBase64(base64String, inputSampleRate, inputChannelCount, isInterleaved = true) {
  return AudioDecoder.getInstance().decodePCMInBase64Instance(base64String, inputSampleRate, inputChannelCount, isInterleaved);
}
