// react-native-audio-api/lib/commonjs/api.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AnalyserNode", {
  enumerable: true,
  get: function () {
    return _AnalyserNode.default;
  }
});
Object.defineProperty(exports, "AudioBuffer", {
  enumerable: true,
  get: function () {
    return _AudioBuffer.default;
  }
});
Object.defineProperty(exports, "AudioBufferQueueSourceNode", {
  enumerable: true,
  get: function () {
    return _AudioBufferQueueSourceNode.default;
  }
});
Object.defineProperty(exports, "AudioBufferSourceNode", {
  enumerable: true,
  get: function () {
    return _AudioBufferSourceNode.default;
  }
});
Object.defineProperty(exports, "AudioContext", {
  enumerable: true,
  get: function () {
    return _AudioContext.default;
  }
});
Object.defineProperty(exports, "AudioDestinationNode", {
  enumerable: true,
  get: function () {
    return _AudioDestinationNode.default;
  }
});
Object.defineProperty(exports, "AudioManager", {
  enumerable: true,
  get: function () {
    return _system.default;
  }
});
Object.defineProperty(exports, "AudioNode", {
  enumerable: true,
  get: function () {
    return _AudioNode.default;
  }
});
Object.defineProperty(exports, "AudioParam", {
  enumerable: true,
  get: function () {
    return _AudioParam.default;
  }
});
Object.defineProperty(exports, "AudioRecorder", {
  enumerable: true,
  get: function () {
    return _AudioRecorder.default;
  }
});
Object.defineProperty(exports, "AudioScheduledSourceNode", {
  enumerable: true,
  get: function () {
    return _AudioScheduledSourceNode.default;
  }
});
Object.defineProperty(exports, "AudioWorkletRuntime", {
  enumerable: true,
  get: function () {
    return _types.AudioWorkletRuntime;
  }
});
Object.defineProperty(exports, "BaseAudioContext", {
  enumerable: true,
  get: function () {
    return _BaseAudioContext.default;
  }
});
Object.defineProperty(exports, "BiquadFilterNode", {
  enumerable: true,
  get: function () {
    return _BiquadFilterNode.default;
  }
});
Object.defineProperty(exports, "BiquadFilterType", {
  enumerable: true,
  get: function () {
    return _types.BiquadFilterType;
  }
});
Object.defineProperty(exports, "ChannelCountMode", {
  enumerable: true,
  get: function () {
    return _types.ChannelCountMode;
  }
});
Object.defineProperty(exports, "ChannelInterpretation", {
  enumerable: true,
  get: function () {
    return _types.ChannelInterpretation;
  }
});
Object.defineProperty(exports, "ConstantSourceNode", {
  enumerable: true,
  get: function () {
    return _ConstantSourceNode.default;
  }
});
Object.defineProperty(exports, "ContextState", {
  enumerable: true,
  get: function () {
    return _types.ContextState;
  }
});
Object.defineProperty(exports, "ConvolverNode", {
  enumerable: true,
  get: function () {
    return _ConvolverNode.default;
  }
});
Object.defineProperty(exports, "GainNode", {
  enumerable: true,
  get: function () {
    return _GainNode.default;
  }
});
Object.defineProperty(exports, "IOSCategory", {
  enumerable: true,
  get: function () {
    return _types2.IOSCategory;
  }
});
Object.defineProperty(exports, "IOSMode", {
  enumerable: true,
  get: function () {
    return _types2.IOSMode;
  }
});
Object.defineProperty(exports, "IOSOption", {
  enumerable: true,
  get: function () {
    return _types2.IOSOption;
  }
});
Object.defineProperty(exports, "IndexSizeError", {
  enumerable: true,
  get: function () {
    return _errors.IndexSizeError;
  }
});
Object.defineProperty(exports, "InvalidAccessError", {
  enumerable: true,
  get: function () {
    return _errors.InvalidAccessError;
  }
});
Object.defineProperty(exports, "InvalidStateError", {
  enumerable: true,
  get: function () {
    return _errors.InvalidStateError;
  }
});
Object.defineProperty(exports, "LockScreenInfo", {
  enumerable: true,
  get: function () {
    return _types2.LockScreenInfo;
  }
});
Object.defineProperty(exports, "MediaState", {
  enumerable: true,
  get: function () {
    return _types2.MediaState;
  }
});
Object.defineProperty(exports, "NotSupportedError", {
  enumerable: true,
  get: function () {
    return _errors.NotSupportedError;
  }
});
Object.defineProperty(exports, "OfflineAudioContext", {
  enumerable: true,
  get: function () {
    return _OfflineAudioContext.default;
  }
});
Object.defineProperty(exports, "OscillatorNode", {
  enumerable: true,
  get: function () {
    return _OscillatorNode.default;
  }
});
Object.defineProperty(exports, "OscillatorType", {
  enumerable: true,
  get: function () {
    return _types.OscillatorType;
  }
});
Object.defineProperty(exports, "PeriodicWaveConstraints", {
  enumerable: true,
  get: function () {
    return _types.PeriodicWaveConstraints;
  }
});
Object.defineProperty(exports, "PermissionStatus", {
  enumerable: true,
  get: function () {
    return _types2.PermissionStatus;
  }
});
Object.defineProperty(exports, "RangeError", {
  enumerable: true,
  get: function () {
    return _errors.RangeError;
  }
});
Object.defineProperty(exports, "RecorderAdapterNode", {
  enumerable: true,
  get: function () {
    return _RecorderAdapterNode.default;
  }
});
Object.defineProperty(exports, "SessionOptions", {
  enumerable: true,
  get: function () {
    return _types2.SessionOptions;
  }
});
Object.defineProperty(exports, "StereoPannerNode", {
  enumerable: true,
  get: function () {
    return _StereoPannerNode.default;
  }
});
Object.defineProperty(exports, "StreamerNode", {
  enumerable: true,
  get: function () {
    return _StreamerNode.default;
  }
});
Object.defineProperty(exports, "WindowType", {
  enumerable: true,
  get: function () {
    return _types.WindowType;
  }
});
Object.defineProperty(exports, "WorkletNode", {
  enumerable: true,
  get: function () {
    return _WorkletNode.default;
  }
});
Object.defineProperty(exports, "WorkletProcessingNode", {
  enumerable: true,
  get: function () {
    return _WorkletProcessingNode.default;
  }
});
Object.defineProperty(exports, "WorkletSourceNode", {
  enumerable: true,
  get: function () {
    return _WorkletSourceNode.default;
  }
});
Object.defineProperty(exports, "changePlaybackSpeed", {
  enumerable: true,
  get: function () {
    return _AudioStretcher.default;
  }
});
Object.defineProperty(exports, "decodeAudioData", {
  enumerable: true,
  get: function () {
    return _AudioDecoder.decodeAudioData;
  }
});
Object.defineProperty(exports, "decodePCMInBase64", {
  enumerable: true,
  get: function () {
    return _AudioDecoder.decodePCMInBase64;
  }
});
Object.defineProperty(exports, "useSystemVolume", {
  enumerable: true,
  get: function () {
    return _useSystemVolume.default;
  }
});
var _specs = require("./specs");
var _WorkletNode = _interopRequireDefault(require("./core/WorkletNode"));
var _WorkletSourceNode = _interopRequireDefault(require("./core/WorkletSourceNode"));
var _WorkletProcessingNode = _interopRequireDefault(require("./core/WorkletProcessingNode"));
var _RecorderAdapterNode = _interopRequireDefault(require("./core/RecorderAdapterNode"));
var _AudioBuffer = _interopRequireDefault(require("./core/AudioBuffer"));
var _AudioBufferSourceNode = _interopRequireDefault(require("./core/AudioBufferSourceNode"));
var _AudioBufferQueueSourceNode = _interopRequireDefault(require("./core/AudioBufferQueueSourceNode"));
var _AudioContext = _interopRequireDefault(require("./core/AudioContext"));
var _OfflineAudioContext = _interopRequireDefault(require("./core/OfflineAudioContext"));
var _AudioDestinationNode = _interopRequireDefault(require("./core/AudioDestinationNode"));
var _AudioNode = _interopRequireDefault(require("./core/AudioNode"));
var _AnalyserNode = _interopRequireDefault(require("./core/AnalyserNode"));
var _AudioParam = _interopRequireDefault(require("./core/AudioParam"));
var _AudioScheduledSourceNode = _interopRequireDefault(require("./core/AudioScheduledSourceNode"));
var _BaseAudioContext = _interopRequireDefault(require("./core/BaseAudioContext"));
var _BiquadFilterNode = _interopRequireDefault(require("./core/BiquadFilterNode"));
var _GainNode = _interopRequireDefault(require("./core/GainNode"));
var _OscillatorNode = _interopRequireDefault(require("./core/OscillatorNode"));
var _StereoPannerNode = _interopRequireDefault(require("./core/StereoPannerNode"));
var _AudioRecorder = _interopRequireDefault(require("./core/AudioRecorder"));
var _StreamerNode = _interopRequireDefault(require("./core/StreamerNode"));
var _ConstantSourceNode = _interopRequireDefault(require("./core/ConstantSourceNode"));
var _system = _interopRequireDefault(require("./system"));
var _ConvolverNode = _interopRequireDefault(require("./core/ConvolverNode"));
var _useSystemVolume = _interopRequireDefault(require("./hooks/useSystemVolume"));
var _AudioDecoder = require("./core/AudioDecoder");
var _AudioStretcher = _interopRequireDefault(require("./core/AudioStretcher"));
var _types = require("./types");
var _types2 = require("./system/types");
var _errors = require("./errors");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/* eslint-disable no-var */

/* eslint-disable no-var */

if (global.createAudioContext == null || global.createOfflineAudioContext == null || global.createAudioRecorder == null || global.createAudioDecoder == null || global.createAudioStretcher == null || global.AudioEventEmitter == null) {
  if (!_specs.NativeAudioAPIModule) {
    throw new Error(`Failed to install react-native-audio-api: The native module could not be found.`);
  }
  _specs.NativeAudioAPIModule.install();
}
