// react-native-audio-api/lib/commonjs/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.assertWorkletsEnabled = assertWorkletsEnabled;
exports.clamp = clamp;
exports.workletsVersion = exports.workletsModule = exports.supportedWorkletsVersions = exports.isWorkletsVersionSupported = void 0;
function clamp(value, min, max) {
  return Math.min(Math.max(value, min), max);
}
let isWorkletsAvailable = false;
let isWorkletsVersionSupported = exports.isWorkletsVersionSupported = false;
let workletsVersion = exports.workletsVersion = 'unknown';
const supportedWorkletsVersions = exports.supportedWorkletsVersions = ['0.6.0', '0.6.1'];
let workletsModule = exports.workletsModule = void 0;
function assertWorkletsEnabled() {
  if (!isWorkletsAvailable) {
    throw new Error('[react-native-audio-api]: Worklets are not available. Please install react-native-worklets to use this feature.');
  } else if (!isWorkletsVersionSupported) {
    throw new Error(`[react-native-audio-api]: Worklets version ${workletsVersion} is not supported.
      Please install react-native-worklets of one of the following versions: [${supportedWorkletsVersions.join(', ')}] to use this feature.`);
  }
}
try {
  exports.workletsModule = workletsModule = require('react-native-worklets');
  const workletsModuleJson = require('react-native-worklets/package.json');
  exports.isWorkletsVersionSupported = isWorkletsVersionSupported = supportedWorkletsVersions.includes(workletsModuleJson.version);
  isWorkletsAvailable = true;
  exports.workletsVersion = workletsVersion = workletsModuleJson.version;
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
} catch (error) {
  isWorkletsAvailable = false;
}
