// react-native-audio-api/lib/module/utils/index.js
"use strict";

export function clamp(value, min, max) {
  return Math.min(Math.max(value, min), max);
}
let isWorkletsAvailable = false;
export let isWorkletsVersionSupported = false;
export let workletsVersion = 'unknown';
export const supportedWorkletsVersions = ['0.6.0', '0.6.1'];
export let workletsModule;
export function assertWorkletsEnabled() {
  if (!isWorkletsAvailable) {
    throw new Error('[react-native-audio-api]: Worklets are not available. Please install react-native-worklets to use this feature.');
  } else if (!isWorkletsVersionSupported) {
    throw new Error(`[react-native-audio-api]: Worklets version ${workletsVersion} is not supported.
      Please install react-native-worklets of one of the following versions: [${supportedWorkletsVersions.join(', ')}] to use this feature.`);
  }
}
try {
  workletsModule = require('react-native-worklets');
  const workletsModuleJson = require('react-native-worklets/package.json');
  isWorkletsVersionSupported = supportedWorkletsVersions.includes(workletsModuleJson.version);
  isWorkletsAvailable = true;
  workletsVersion = workletsModuleJson.version;
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
} catch (error) {
  isWorkletsAvailable = false;
}
